### PR TITLE
(Fix) 03-3450: Delete modal not closing and package units not refreshing after deleting.

### PR DESCRIPTION
**Pull request by hadijahkyampeire**
## Requirements
- [x] This PR has a title that briefly describes the work done including a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) type prefix and a Jira ticket number if applicable. See existing PR titles for inspiration.

#### For changes to apps
- [ ]  My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide) and [**design documentation**](https://zeroheight.com/23a080e38/p/880723-introduction).

#### If applicable
- [ ] My work includes tests or is validated by existing tests.

## Summary
- Add combox placeholder
- Close the delete module after delete
- Refresh the packingUnits after deleting

## Screenshots

https://github.com/openmrs/openmrs-esm-stock-management/assets/30952856/d188a7d0-136b-4e13-a2c4-35d460f0f4a0

## Related Issue
- https://openmrs.atlassian.net/browse/O3-3450

### CODE REVIEW

File: src/stock-items/add-stock-item/stock-item-units-edit/stock-item-units-edit.component.tsx [hunk 1]
```diff
@@ -120,7 +120,10 @@ const StockItemUnitsEdit: React.FC<StockItemUnitsEditProps> = ({
         name="purchasePriceUoMUuid"
         controllerName="purchasePriceUoMUuid"
         control={control}
-        title={t("purchasePricePackagingUnit", "Purchase price packaging unit")}
+        title={t(
+          "purchasePricePackagingUnit",
```
Comment (ojwanganto) [inline src/stock-items/add-stock-item/stock-item-units-edit/stock-item-units-edit.component.tsx:124]: Do you know what this is supposed to mean?

**Comment by ojwanganto** [comment]
Thanks, @hadijahkyampeire. I wanted to confirm if this 'delete' ideally means voiding and not purging. This is to help preserve the integrity of any references to it. Thanks

File: src/stock-items/add-stock-item/stock-item-units-edit/stock-item-units-edit.component.tsx (hunk 1 above)
Comment (hadijahkyampeire) [reply src/stock-items/add-stock-item/stock-item-units-edit/stock-item-units-edit.component.tsx:124]: The title of that field

**Comment by hadijahkyampeire** [comment]
@ojwanganto I think that would be a question for a backend dev maybe @slubwama, but on the frontend I can not tell whether it is a permanent delete or void, all I know is that we have an endpoint that deletes and that packaging unit disappears from the list of packaging units.

**Review by jabahum — APPROVED** [review]
LGTM @hadijahkyampeire

**Pull request merged**